key_gen.go

```diff
@@ -54,12 +54,22 @@ func KeyForPrefixedStringMap(
 	prefix string,
 	stringMap map[string]string,
 ) string {
+	return keyForPrefixedStringMaps(prefix, stringMap)
+}
+
+// keyForPrefixedStringMaps generates a unique key for a prefix and a series
+// of maps containing tags.
+//
+// If a key occurs in multiple maps, keys on the right take precedence.
+func keyForPrefixedStringMaps(prefix string, maps ...map[string]string) string {
```

Comment:
if you wanted to take this a step further, it would be even more performant to keep slices of tags (key/value pairs) and then just merging those with user-provided tag map on scope creation.

Reply:
Yeah, there's more room for optimization on this code path. I may explore it more.
This was just a low hanging fruit we ran into during a conversation on Friday.